spec/requests/api/v3/user/userlock_resource_spec.rb

```diff
@@ -0,0 +1,122 @@
+#-- copyright
+# OpenProject is a project management system.
+# Copyright (C) 2012-2015 the OpenProject Foundation (OPF)
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See doc/COPYRIGHT.rdoc for more details.
+#++
+
+require 'spec_helper'
+require 'rack/test'
+
+describe 'API v3 UserLock resource', type: :request do
+  include Rack::Test::Methods
+
+  let(:current_user) { FactoryGirl.build_stubbed(:user) }
+  let(:user) { FactoryGirl.create(:user, status: User::STATUSES[:active]) }
+  let(:model) { ::API::V3::Users::UserModel.new(user) }
+  let(:representer) { ::API::V3::Users::UserRepresenter.new(model) }
+  let(:lock_path) { "/api/v3/users/#{user.id}/lock" }
+  subject(:response) { last_response }
+
+  describe '#post' do
+    before do
+      allow(User).to receive(:current).and_return current_user
+      post lock_path
+      # lock manually
+      user.lock
+    end
+
+    # Locking is only available for admins
+    context 'when logged in as admin' do
+      let(:current_user) { FactoryGirl.build_stubbed(:admin) }
+
+      context 'user account can be locked' do
+        it 'should respond with 200' do
+          expect(subject.status).to eq(200)
+        end
+
+        it 'should respond with an updated lock status in the user model' do
+          expect(parse_json(subject.body, 'status')).to eq 'locked'
+        end
+      end
+
+      context 'user account is incompatible' do
+        let(:user) { FactoryGirl.create(:user,
+          status: User::STATUSES[:registered]) }
```

Comment:
Align the parameters of a method call if they span more than one line.<br>Expression at 66, 48 should be on its own line.